### DISCUSSION TITLE
Suggestion for HTTP Headers configuration?

### DISCUSSION THREAD

**Discussion by gastaldi** [quarkusio/quarkus#25139] [Polls]
The existing [Additional HTTP Headers](https://quarkus.io/guides/http-reference#additional-http-headers) guide does not let you specify different header values for different paths (see the original issue [here](https://github.com/quarkusio/quarkus/issues/25047)) .

The following PR introduces a new way of configuring fixed HTTP headers for a given path: 
 
- https://github.com/quarkusio/quarkus/pull/25056

However, naming is hard, therefore, we'd like to know your opinion on a good configuration option for this feature.

**Comment by Postremus** [comment]
quarkus.http.response.headers."/*".Cache-control=none

**Reply by gastaldi** [reply]
Interesting. I think the only problem with this configuration is that you also need to configure ordering and the allowed HTTP methods for each entry.

Maybe it could work like this?  

```properties
quarkus.http.response.headers."/*".Cache-control=none
quarkus.http.response.headers."/*".Cache-control.order=1
quarkus.http.response.headers."/*".Cache-control.methods=GET,POST
```

**Reply by Ladicek** [reply]
You could make it `quarkus.http.response.headers."/*".Cache-control.value=none` for uniformity.

(My personal vote would be this, without the `response` word :-) )

**Comment by dmlloyd** [comment]
Another alternative - give the matching nature of the path - would be something like this:

```properties
quarkus.http.by-path.my-app.header.Cache-control=none
quarkus.http.by-path.my-app.matches=/app/*|/other-path|/blah
```

This gets the matching part out of the configuration key, which in turn makes it possible to override the path matching using a system property e.g. `-Dquarkus.http.by-path.my-app.matches=/foobar` which is a lot harder when it is part of the key.

In other words, since the pattern is part of the configuration, maybe it doesn't actually belong in the key after all.

**Reply by gastaldi** [reply]
That makes sense, I'll give it a try
